Panic when upgrading from 1.11.12 to 1.11.14 with disable-endpoint-crd: true

### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

Standard upgrade from 1.11.12 to 1.11.14. The cilium-agent crashed-looped:
```
...
level=info msg="Finished regenerating restored endpoints" regenerated=5 subsys=daemon total=5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x22dd54d]

goroutine 996 [running]:
github.com/cilium/cilium/daemon/cmd.(*Daemon).cleanStaleCEPs(0xc0000e4c80, {0x2f8e088, 0xc00007c020}, {0x2f2b040, 0xc0003a40e0}, {0x2fcf838, 0xc0007fc3a0}, 0x0)
  /go/src/github.com/cilium/cilium/daemon/cmd/ciliumendpoints.go:34 +0xcd
github.com/cilium/cilium/daemon/cmd.runDaemon.func2()
  /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1770 +0xa5
created by github.com/cilium/cilium/daemon/cmd.runDaemon
  /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1759 +0x1473

      Exit Code:    2
```

### Cilium Version

1.11.14

### Kernel Version

5.15.43-flatcar

### Kubernetes Version

v1.23.15

### Sysdump

_No response_

### Relevant log output

```shell
level=info msg="Rewrote endpoint BPF program" containerID=70bbd5244d datapathPolicyRevision=0 desiredPolicyRevision=14 endpointID=2350 identity=19528 ipv4=172.16.11.219 ipv6= k8sPodName=kube-system/fluent-bit-splunk-es-zw8lb subsys=endpoint
level=info msg="Restored endpoint" endpointID=2350 ipAddr="[172.16.11.219 ]" subsys=endpoint
level=info msg="Restored endpoint" endpointID=683 ipAddr="[172.16.11.141 ]" subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID=94f47aad58 datapathPolicyRevision=0 desiredPolicyRevision=14 endpointID=3192 identity=5237 ipv4=172.16.11.110 ipv6= k8sPodName=monitoring/cert-exporter-wd478 subsys=endpoint
level=info msg="Rewrote endpoint BPF program" containerID=8cd459e8a1 datapathPolicyRevision=0 desiredPolicyRevision=14 endpointID=27 identity=27481 ipv4=172.16.11.102 ipv6= k8sPodName=kube-system/fluent-bit-journald-946bn subsys=endpoint
level=info msg="Restored endpoint" endpointID=3192 ipAddr="[172.16.11.110 ]" subsys=endpoint
level=info msg="Restored endpoint" endpointID=27 ipAddr="[172.16.11.102 ]" subsys=endpoint
level=info msg="Finished regenerating restored endpoints" regenerated=5 subsys=daemon total=5
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x22dd54d]

goroutine 996 [running]:
github.com/cilium/cilium/daemon/cmd.(*Daemon).cleanStaleCEPs(0xc0000e4c80, {0x2f8e088, 0xc00007c020}, {0x2f2b040, 0xc0003a40e0}, {0x2fcf838, 0xc0007fc3a0}, 0x0)
  /go/src/github.com/cilium/cilium/daemon/cmd/ciliumendpoints.go:34 +0xcd
github.com/cilium/cilium/daemon/cmd.runDaemon.func2()
  /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1770 +0xa5
created by github.com/cilium/cilium/daemon/cmd.runDaemon
  /go/src/github.com/cilium/cilium/daemon/cmd/daemon_main.go:1759 +0x1473
```


### Anything else?

Seems like PR #20350 doesn't take `disable-endpoint-crd: "true"` into account.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/daemon/cmd/ciliumendpoints.go
+++ b/daemon/cmd/ciliumendpoints.go
@@ -31,6 +31,9 @@ func (d *Daemon) cleanStaleCEPs(ctx context.Context, eps localEndpointCache, cil
 		crdType = "ciliumendpointslice"
 	}
 	indexer := d.k8sWatcher.GetIndexer(crdType)
+	if indexer == nil {
+		return fmt.Errorf("%s indexer was nil", crdType)
+	}
 	objs, err := indexer.ByIndex("localNode", node.GetCiliumEndpointNodeIP())
 	if err != nil {
 		return fmt.Errorf("could not get %s objects from localNode indexer: %w", crdType, err)

--- a/daemon/cmd/daemon_main.go
+++ b/daemon/cmd/daemon_main.go
@@ -1784,7 +1784,9 @@ func runDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *daem
 			<-restoreComplete
 		}
 
-		if params.Clientset.IsEnabled() && option.Config.EnableStaleCiliumEndpointCleanup {
+		// Only attempt CEP cleanup if cilium endpoint CRD is not disabled, otherwise the cep/ces
+		// watchers/indexers will not be initialized.
+		if params.Clientset.IsEnabled() && option.Config.EnableStaleCiliumEndpointCleanup && !option.Config.DisableCiliumEndpointCRD {
 			// Use restored endpoints to delete local CiliumEndpoints which are not in the restored endpoint cache.
 			// This will clear out any CiliumEndpoints that may be stale.
 			// Likely causes for this are Pods having their init container restarted or the node being restarted.

